server/index.js

```diff
@@ -150,6 +150,41 @@ app.post('/api/cart/:productId', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.post('/api/orders', (req, res, next) => {
+  const cartId = req.session.cartId;
+  if (!cartId) {
+    res.status(400).json({ error: 'no cart id found.' });
+  }
+  const { name, creditCard, shippingAddress } = req.body;
+  const errors = {};
+  if (!name) {
+    errors.name = 'missing or invalid name.';
+  }
+  if (!creditCard) {
+    errors.creditCard = 'missing or invalid credit card number.';
+  }
+  if (!shippingAddress) {
+    errors.shippingAddress = 'missing or invalid shipping address.';
+  }
+  if (Object.keys(errors).length) {
+    res.status(400).json(errors);
+  }
+
+  const sql = `
+  INSERT INTO "orders" ("name", "creditCard", "shippingAddress", "cartId")
+  VALUES ($1, $2, $3, $4)
+  RETURNING "orderId", "createdAt", "name", "creditCard", "shippingAddress"
+  `;
+  const values = [name, creditCard, shippingAddress, cartId];
+
+  db.query(sql, values)
+    .then(result => {
+      delete req.session.cartId;
+      res.status(201).json(result.rows);
```

Comment:
Typically, when a single record is successfully created, we do not send it back in an array.